apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/SwapWebView.tsx

```diff
@@ -174,6 +190,32 @@ const SwapWebView = ({ manifest, swapState, liveAppUnavailable }: SwapWebProps)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [webviewState.url]);
 
+  const hashString = useMemo(() => {
+    const params = {
+      from: fromCurrency,
+      to: toCurrency,
+      amountFrom: swapState?.fromAmount,
+      addressFrom: addressFrom,
+      addressTo: addressTo,
+      networkFees: swapState?.estimatedFees,
+      networkFeesCurrency: fromCurrency,
+    };
+
+    const hash = Object.entries(params)
+      .filter(([, value]) => Boolean(value)) // Filter out null or undefined values
+      .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(value || "")}`)
+      .join("&");
```

Comment:
For this, even though it is for a hash we can use the `URLSearchParams` class to generate the hash.

Reply:
fixed